Framework/CurveFitting/src/IMWDomainCreator.cpp

```diff
@@ -186,6 +188,14 @@ std::pair<size_t, size_t> IMWDomainCreator::getXInterval() const {
     }
   }
 
+  // Check whether the fitting interval defined by StartX and EndX is 0.
+  // This occurs when StartX and EndX are both less than the minimum workspace
+  // x-value or greater than the maximum workspace x-value.
+  if (to - from == 0) {
```

Comment:
To and from are iterators from finding given StartX and EndX values in a workspace X range, respectively. StartX and EndX are swapped if StartX > EndX. Therefore not possible for to < from (StartX will always appear before or in the same place as EndX).